lib/ansible/playbook/play_context.py

```diff
@@ -434,10 +434,15 @@ def set_task_and_variable_override(self, task, variables, templar):
         # set become defaults if not previouslly set
         task.set_become_defaults(new_info.become, new_info.become_method, new_info.become_user)
 
-        # have always_run override check mode
         if task.always_run:
+            display.deprecated("always_run is deprecated. Use check_mode = no instead.", version="2.4", removed=False)
```

Comment:
I would change the message to just: `always_run is deprecated. Use check_mode instead`